include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -1324,6 +1324,292 @@ namespace Differentiation
+     *     // This is also taken care of by the ADHelper, so this step could
+     *     // be skipped.
+     *     cell_rhs.reinit(n_dependent_variables);
+     *     cell_matrix.reinit(n_independent_variables,n_dependent_variables);
+     *
+     *     // This next code block corresponds to the "recording" phase, where
+     *     // the operations performed with the AD numbers are tracked and
+     *     // differentiation is performed.
+     *     {
+     *       // First, we set the values for all DoFs.
+     *       ad_helper.register_dof_values(solution, local_dof_indices);
+     *
+     *       // Then we get the complete set of degree-of-freedom values as
+     *       // represented by auto-differentiable numbers. The operations
+     *       // performed with these variables are tracked by the AD library
+     *       // from this point until stop_recording_operations() is called.
+     *       const std::vector<ADNumberType> dof_values_ad
+     *         = ad_helper.get_sensitive_dof_values();
+     *
+     *       // Then we do some problem specific tasks, the first being to
+     *       // compute all values, gradients etc. based on sensitive AD DoF
+     *       // values. Here we are fetching the displacement gradients at each
+     *       // quadrature point, as well as the gradients of the magnetic
+     *       // scalar potential field.
+     *       std::vector<Tensor<2, dim, ADNumberType>> Grad_u(
+     *         n_q_points, Tensor<2, dim, ADNumberType>());
+     *       std::vector<Tensor<1, dim, ADNumberType>> Grad_msp(
+     *         n_q_points, Tensor<1, dim, ADNumberType>());
+     *       fe_values[u_fe].get_function_gradients_from_local_dof_values(
+     *         dof_values_ad, Grad_u);
+     *       fe_values[msp_fe].get_function_gradients_from_local_dof_values(
+     *         dof_values_ad, Grad_msp);
+     *
+     *       // This variable stores the cell residual vector contributions.
+     *       // IMPORTANT: Note that each entry is hand-initialized with a value
+     *       // of zero. This is a highly recommended practise, as some AD
+     *       // numbers appear not to safely initialize their internal data
+     *       // structures.
+     *       std::vector<ADNumberType> residual_ad (
+     *         n_dependent_variables, ADNumberType(0.0));
+     *
+     *       // Compute the cell total residual
+     *       //   = (internal + external) contributions
+     *       for (unsigned int q_point = 0; q_point < n_q_points; ++q_point)
+     *       {
+     *         // Calculate the deformation gradient and magnetic field at this
+     *         // quadrature point
+     *         const Tensor<2, dim, ADNumberType> F =
+     *           unit_symmetric_tensor<dim>() + Grad_u[q_point];
+     *         const Tensor<1, dim, ADNumberType> H = -Grad_msp[q_point];
+     *         Assert(numbers::value_is_greater_than(determinant(F), 0.0),
+     *                ExcMessage("Negative jacobian detected!"));
```

Comment:
-> Jacobian

Is this the right term? (I don't know -- not my field :-) ) Maybe just say "Negative determinant of the deformation gradient detected"